### ISSUE TITLE
MODULE_PATH

### ISSUE THREAD

**Issue by yantrab** [nestjs/nest#1520]
Hi,

There is something like Reflect.getMetadata(MODULE_PATH, metatype) in routes-resolver.ts
but there is no Reflect.defineMetadata with this constant , so I  wonder from where this metadata came from, there is an options to set the module path?

Thanks.

**Comment by BrunnerLivio** [comment]
Maybe check out the source code of [nest-router](https://github.com/shekohex/nest-router/blob/master/src/router.module.ts)

The author set the module path in `RouterModule#buildPathMap`

**Comment by kamilmysliwiec** [comment]
Worth adding - router module functionality should be very likely added to the core package and brought under official nestjs organization.

**Comment by yantrab** [comment]
@BrunnerLivio thanks.

There is an options to get the path of all controllers ?

**Comment by yantrab** [comment]
@kamilmysliwiec i meant to file path. i wont to build client generator, according to the controller statements.
i don't wont to using swagger, the models locateted in shared folder, and i wont to use them directly.

**Comment by victordidenko** [comment]
Wanted to create an issue, but found this one.
I'm not sure, but shouldn't be there
https://github.com/nestjs/nest/blob/85d7384b5e2b486af42d13c7940ee2e3fa63256d/packages/core/router/routes-resolver.ts#L45
```
path = path ? basePath + path : basePath;
```
instead of
```
path = path ? path + basePath : basePath;
```
? Because `basePath` is global prefix here https://github.com/nestjs/nest/blob/bf04704e11c6a6ffc15e1188a450c77892c04792/packages/core/nest-application.ts#L193

**Comment by victordidenko** [comment]
My last message was fixed in #1594
